ci/setup_cache.sh

```diff
@@ -47,6 +36,14 @@ if [[ -n "${BAZEL_REMOTE_CACHE}" ]]; then
     if [[ -z "${ENVOY_RBE}" ]]; then
         export BAZEL_BUILD_EXTRA_OPTIONS+=" --jobs=HOST_CPUS*.99 --remote_timeout=600"
         echo "using local build cache."
+        # Normalize branches - `release/vX.xx`, `vX.xx`, `vX.xx.x` -> `vX.xx`
+        BRANCH_NAME="$(echo "${CI_TARGET_BRANCH}" | cut -d/ -f2 | cut -d. -f-2)"
+        if [[ "$BRANCH_NAME" == "merge" ]]; then
+            # Manually run PR commit - there is no easy way of telling which branch
+            # it is, so just set it to `main` - otherwise it tries to cache as `branch/merge`
+            BRANCH_NAME=main
+        fi
+        BAZEL_REMOTE_INSTANCE="branch/${BRANCH_NAME}"
```

Comment:
does this change make sense? And I think there is a behavior change:

```
    if [[ -n "${BAZEL_REMOTE_INSTANCE}" ]]; then
        export BAZEL_BUILD_EXTRA_OPTIONS+=" --remote_instance_name=${BAZEL_REMOTE_INSTANCE}"
        echo "instance_name: ${BAZEL_REMOTE_INSTANCE}."
    fi
```
will be evaludated before the `BAZEL_REMOTE_INSTANCE` is set? is this expected?

Reply:
`BAZEL_REMOTE_INSTANCE` is only set if `ENVOY_RBE` is so it should be the same - this could probably be improved further, but im ~confident this change doesnt actually change anything (just consolidates vars)

Reply:
see https://github.com/envoyproxy/envoy/blob/809eaf83e1df95f7922a482d67f2e81ad1dd9443/.azure-pipelines/bazel.yml#L190-L212

Reply:
it seems this confidence was misplaced - its broken the branch caching

ill look further

Reply:
huge apology - you were absolutely correct - the order was incorrect